<launch>
    <node name="dynamixel_manager" pkg="dynamixel_controllers" type="controller_manager.py" required="true" output="screen">
        <rosparam>
            namespace: dxl_manager
            serial_ports:
                actuator_port:
                    port_name: "/dev/ttyUSB1"
                    baud_rate: 57600
                    min_motor_id: 1
                    max_motor_id: 50
                    update_rate: 15
        </rosparam>
    </node>
    
	    <!-- Load controller configuration to parameter server -->
    <rosparam file="$(find robot_connect)/launch/config/dynamixel_joint_controllers.yaml" command="load"/>
    
    <!-- start specified joint controllers -->
    <node name="dynamixel_controller_spawner" pkg="dynamixel_controllers" type="controller_spawner.py"
          args="--manager=dxl_manager
                --port=actuator_port
                --type=simple
                pan_kinect
                tilt_kinect
                mark44_1
                mark44_2
                mark44_3
		mark43_pris
                gripper
                joint1
                joint2
                joint3"
          output="screen"/>
          
    <!--node name="tilt_laser_control" pkg="robot_connect" type="tilt_laser_control" output="screen"/-->
    <!--node name="pan_tilt_control" pkg="robot_connect" type="pan_tilt_control.py"/-->
    <!--node pkg="robot_connect" type="kinect_preprocess" name="kinect_preprocess" output="screen"/-->
</launch>
